asyncapi/identity.server/dispatcher_service.bal

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2025 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerina/lang.regexp;
+import ballerinax/asyncapi.native.handler;
+import ballerina/crypto;
+
+service class DispatcherService {
+    *http:Service;
+    private map<GenericServiceType> services = {};
+    private handler:NativeHandler nativeHandler = new ();
+    private ListenerConfig listenerConfig;
+
+    public function init(ListenerConfig listenerConfig) {
+        self.listenerConfig = listenerConfig;
+    }
+
+    isolated function addServiceRef(string serviceType, GenericServiceType genericService) returns error? {
+        if (self.services.hasKey(serviceType)) {
+            return error("Service of type " + serviceType + " has already been attached");
+        }
+        self.services[serviceType] = genericService;
+    }
+
+    isolated function removeServiceRef(string serviceType) returns error? {
+        if (!self.services.hasKey(serviceType)) {
+            return error("Cannot detach the service of type " + serviceType +
+            ". Service has not been attached to the listener before");
+        }
+        _ = self.services.remove(serviceType);
+    }
+
+    // We are not using the (@http:payload GenericEventWrapperEvent g) notation because of a bug in Ballerina.
+    // Issue: https://github.com/ballerina-platform/ballerina-lang/issues/32859
```

Comment:
This issue is now closed, may be able to refactor.

Reply:
As discussed offline, will incorporate this change later